test/distributed/test_hooks.py

```diff
@@ -0,0 +1,270 @@
+# Owner(s): ["oncall: distributed"]
+
+import os
+import sys
+import tempfile
+import threading
+from functools import partial, wraps
+
+import torch
+import torch.distributed as dist
+import torch.distributed._hooks as dhooks
+
+if not dist.is_available():
+    print("torch.distributed not available, skipping tests", file=sys.stderr)
+    sys.exit(0)
```

Comment:
This is a bit unconventional, but I see you following a pattern, will submit a followup PR later (i.e. python -c "import test_hooks;print('Hello World')" should work, but it would not in your case.